### ISSUE TITLE
Sum linked transactions

### ISSUE THREAD

**Issue by hamuz** [firefly-iii/firefly-iii#1749]
**Description**
Please describe your feature request:

- I would like Firefly III to sum all linked transactions

**Solution**
Describe what your feature would add to Firefly III.

Easier spotting of errors when linked transactions exceed the transaction amount.

**What are alternatives?**
Please describe what alternatives currently exist.

Manual calculator.

**Additional context**

![image](https://user-images.githubusercontent.com/550499/46255473-0c03f000-c4a6-11e8-9b55-922fc986707d.png)

Useful in use case of credit card statements, which can be recreated in FireFly.
Corresponding transactions of credit card are linked to the transfer (payment) from bank.
The sum should match to the amount. Help to detect missing or excessive transactions.

**Comment by hamuz** [comment]
Added screenshot to help with visualization.

**Comment by JC5** [comment]
Not a bad idea. Will write it down. :+1:

**Comment by hamuz** [comment]
Can you outline what needs to be changed/added to implement this? I can try to do it myself...

**Comment by JC5** [comment]
It's actually pretty tricky when you think about it. Worst case is two different currencies per linked transaction (default + foreign) and they could all be different.

The current code focuses on retrieving the links ([one](https://github.com/firefly-iii/firefly-iii/blob/master/app/Http/Controllers/TransactionController.php#L229), [two](https://github.com/firefly-iii/firefly-iii/blob/master/app/Repositories/LinkType/LinkTypeRepository.php#L190)), filters out deleted journals and then uses the pretty generic `journalTotalAmount` to show the amount ([one](https://github.com/firefly-iii/firefly-iii/blob/master/resources/views/transactions/show.twig#L289), [two](https://github.com/firefly-iii/firefly-iii/blob/master/resources/views/transactions/show.twig#L312), [three](https://github.com/firefly-iii/firefly-iii/blob/master/app/Support/Twig/Extension/TransactionJournal.php#L98)).

To make it easier, these could also be split, and splits can have multiple currencies as well.

I have some ideas what you could do but it is pretty complex. You need a function in linktyperepository that returns the transactions in one array and the relevant sums in another one.

**Comment by JC5** [comment]
Yes, but that's just a basic link. Worst case scenario is something like this:

![screenshot 2018-09-30 at 19 10 32](https://user-images.githubusercontent.com/5889984/46260330-cdd9f100-c4e4-11e8-9c9c-9540961ed7c3.png)

**Comment by hamuz** [comment]
And in the new UI would look like this:

![image](https://user-images.githubusercontent.com/550499/46260567-5d38d180-c4f0-11e8-942d-8b69624d3033.png)

```
<div class="box">
    <div class="box-header with-border">
        <h3 class="box-title">This transfer is (partially) paid for by</h3>
        <div class="box-tools pull-right">
            <span class="label label-default">
                <span class="text-danger">€-145.70</span> /
                <span class="text-danger">£-15.00</span> /
                <span class="text-danger">$-11.21</span> /
                <span class="text-danger">Ft-12,342.00</span></span>
        </div>
    </div>
    <div class="box-body table-responsive no-padding">
        <table class="table table-hover">
            <tbody>
                <tr>
                    <td>
                        <div class="btn-group btn-group-xs"><a href="https://demo.firefly-iii.org/transactions/link/delete/3" class="btn btn-danger"><i class="fa fa-trash"></i></a> <a href="https://demo.firefly-iii.org/transactions/link/switch/3" class="btn btn-default"><i class="fa fa-fw fa-arrows-h"></i></a></div>
                    </td>
                    <td>
                        <a href="https://demo.firefly-iii.org/transactions/show/489">
                            #489: Saving money for December</a>
                        (<span class="text-success">€140.00</span>)
                    </td>
                </tr>
                <tr>
                    <td>
                        <div class="btn-group btn-group-xs"><a href="https://demo.firefly-iii.org/transactions/link/delete/3" class="btn btn-danger"><i class="fa fa-trash"></i></a> <a href="https://demo.firefly-iii.org/transactions/link/switch/3" class="btn btn-default"><i class="fa fa-fw fa-arrows-h"></i></a></div>
                    </td>
                    <td>
                        <a href="https://demo.firefly-iii.org/transactions/show/489">
                            ...</a>
                        (<span class="text-success">...</span>)
                    </td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
```

**Comment by JC5** [comment]
Why not a short table in the footer or something?

**Comment by hamuz** [comment]
Can you post screenshot of your mockup so we can compare benefits in each case?

**Comment by JC5** [comment]
Something like this.

![screenshot 2018-09-30 at 20 12 03](https://user-images.githubusercontent.com/5889984/46260944-1e554c80-c4ed-11e8-889c-298150f868b2.png)

**Comment by hamuz** [comment]
1. How would the sum be calculated in your suggestion if different link types are used?
In my suggestion, different link type would be in a separate box.
2. Why do you prefer different currencies on separate lines rather than in one?

**Comment by JC5** [comment]
1. Why would you separate the link type sums?
2. Because it's more clear.

**Comment by hamuz** [comment]
If A pays for B, and B pays for C, and we are looking at B,
summing what A pays to B with what B pays to C makes no sense.

**Comment by JC5** [comment]
But then you would see the amounts are equal anyway?

**Comment by hamuz** [comment]
As your link type description says, it partly paid. The case I discussed is a special cause of same link in opposite directions.
What if A pays for B, and B is related to C, or B to C is custom link? Again, summing different links makes no sense. Even summing different directions of same link, doesn't makes sense.
And subtracting opposite directions, might not always make sense (in custom scenario for example, in "relates to" link type, etc.).
This is why I suggested different box for each link type and direction combination, with it's own sum.
Although most of them will not appear, unless used.
